dll/win32/shell32/CDefView.cpp

```diff
@@ -607,11 +619,29 @@ void CDefView::UpdateStatusbar()
             pIcon = (LPARAM)m_hMyComputerIcon;
         }
 
+        if (dwThreadId != m_uUpdateStatusbarThreadId)
+            return;
+
         m_pShellBrowser->SendControlMsg(FCW_STATUS, SB_SETICON, 2, pIcon, &lResult);
         m_pShellBrowser->SendControlMsg(FCW_STATUS, SB_SETTEXT, 2, (LPARAM)szPartText, &lResult);
     }
 }
 
+static unsigned __stdcall UpdateStatusbarProc(void *args)
+{
+    static_cast<CDefView*>(args)->UpdateStatusbarInner();
+    return 0;
+}
+
+void CDefView::UpdateStatusbar()
+{
+    HANDLE hOldThread = m_hUpdateStatusbarThread;
+    m_hUpdateStatusbarThread = reinterpret_cast<HANDLE>(
+        _beginthreadex(NULL, 0, UpdateStatusbarProc, this, 0, &m_uUpdateStatusbarThreadId));
```

Comment:
Does shell32 already use that function elsewhere to create threads?

Reply:
There is `SHCreateThread` function in `shlwapi`, but it has no thread id paramter.

Reply:
Do we need to keep a thread ID around for this PR functionality, or is it sufficient to keep the handle?

Reply:
Using IDs is better in mechanism. That will distinguish threads.

Reply:
imo thread handles better distinguish them: they are guaranteed to be unique and you can control the thread by using handles, not by using IDs.

Reply:
Used thread handles. Dismissed thread IDs.